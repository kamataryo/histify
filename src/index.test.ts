import Histify from "./";

test("should be sorted", () => {
  const histify = new Histify("1,2,1,4,3,6");
  expect(histify.samples).toEqual([1, 1, 2, 3, 4, 6]);
});

test("classes", () => {
  const histify = new Histify("1,2,3,4,5,6,7");
  const classes = histify.classes();
  expect(classes[0]).toEqual(1.6);
  expect(classes[9]).toEqual(7);
});

test("frequencies", () => {
  const histify = new Histify("1,1,3,3,3,3,3,3,8,10");
  const frequencies = histify.frequency();
  expect(frequencies).toEqual([2, 0, 6, 0, 0, 0, 0, 1, 0, 1]);
});

test("summary", () => {
  const histify = new Histify("1,1,12,2,2,2,2,2,2,2,4,5,6,7,9,8,9");
  expect(histify.summary()).toMatchSnapshot();
  console.log(histify.summary());
});
