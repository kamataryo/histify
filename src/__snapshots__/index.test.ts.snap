// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary 1`] = `
"samples: 17
average: 4.470588235294118
variance: 10.955017301038064
STDEV: 3.3098364462671057
STDEPV: 3.4117013178695674

 to 2.1 ┃ *************************************************************************************************
 to 3.2 ┃ 
 to 4.3 ┃ **********
 to 5.4 ┃ **********
 to 6.5 ┃ **********
 to 7.6 ┃ **********
 to 8.7 ┃ **********
 to 9.8 ┃ *********************
to 10.9 ┃ 
  to 12 ┃ **********
━━━━━━━━╋━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
          0                                                                                               9
  "
`;
